### PR TITLE
Added Trie in C++

### PR DESCRIPTION
**Pull request by devanshiv123**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added Trie implementation in C++

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#849

### CODE REVIEW

**Comment by devanshiv123** [comment]
Kindly Review this PR @iamrajiv @plazzy99 @SKAUL05 @Abhijit2505 @siddharth25pandey

File: C-Plus-Plus/ds/Trie_implementation.cpp [hunk 1]
```diff
@@ -0,0 +1,159 @@
+//Tries are tree based structures used for efficient retrieval of key form huge set of strings
```
Comment (iamrajiv) [inline C-Plus-Plus/ds/Trie_implementation.cpp]: @devanshiv123 make all single-line comments int one multiline comment.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@devanshiv123 remove indentations and unnecessary line breaks.

File: C-Plus-Plus/ds/Trie_implementation.cpp (hunk 1 above)
Comment (devanshiv123) [reply C-Plus-Plus/ds/Trie_implementation.cpp]: @iamrajiv I have made the changes. I have removed unnecessary line breaks and uneccessary indents but i havn't removed all indentations as i thought that they were necessary. Please tell if they are also to be removed.

**Review by Abhijit2505 — APPROVED** [review]
Looking Good To Me

**Review by iamrajiv — APPROVED** [review]

**Review by atarax665 — APPROVED** [review]

**Pull request merged**